File: dataset/human/python/file_4125.py
import math
n = int(input())
n1 = n
if n == 1:
    print(0)
else:
    a = [0]*(int(math.sqrt(n)) + 1)
    a[1] = 1
    ans = 0
    for i in range(2, int(math.sqrt(n)) + 1):
        if a[i] == 0:
            for j in range(i+i, int(math.sqrt(n)) + 1, i):
                a[j] = 1
    for i in range(2, len(a)):
        if a[i] == 0:
            k = 1
            s = i
            while n % s == 0:
                # print(s)
                ans += 1
                n //= s
                s *= i
            while n % i == 0:
                n //= i
    # print(n)
    f = 1
    for j in range(2, int(math.sqrt(n)) + 2):
        if n % j == 0:
            f = 0
            break
    if f and n != 1:
        ans += f
    # print(n1 // i)

    if ans == 0:
        print(1)
    else:
        print(ans)
    # print(len('997764507000'))

File: dataset/human/cpp/file_4125.cpp
#include<iostream>
#include<math.h>
#include<algorithm>

using namespace std;

int main(void)
{
	long long xs,ys,xt,yt;
	cin >> xs >> ys >> xt >> yt;
	long long n;
	cin >> n;
	long long x[n+2],y[n+2],r[n+2];
	for(int i=1;i<=n;i++)
		cin >> x[i] >> y[i] >> r[i];
	x[0]=xs;
	y[0]=ys;
	r[0]=0;
	x[n+1]=xt;
	y[n+1]=yt;
	r[n+1]=0;

	long double dis[n+2][n+2];
	for(int i=1;i<=n+2;i++)
	{
		for(int j=1;j<=n+2;j++)
		{
			long long disx=x[i-1]-x[j-1];
			long long disy=y[i-1]-y[j-1];
			dis[i-1][j-1]=(sqrt(disx*disx+disy*disy)-r[i-1]-r[j-1]);
			if(dis[i-1][j-1]<0)
				dis[i-1][j-1]=0;
//cout << dis[i-1][j-1] << endl;
		}
	}

	bool search[n+2];
	for(int i=1;i<=n+2;i++)
		search[i-1]=false;
	search[0]=true;
	long double mingo[n+2];
	for(int i=1;i<=n+2;i++)
		mingo[i-1]=10000000010;
	mingo[0]=0;

	int now=1;
	

	//dijkstra
	while(true)
	{
		if(now==n+2)
		{
			break;
		}
		long double nowmin = 10000000010;
		int next=0;
		for(int i=1;i<=n+2;i++)
		{
			if(search[i-1]==true)
				continue;
			mingo[i-1]=min(mingo[i-1],mingo[now-1]+dis[now-1][i-1]);
			if(mingo[i-1]<nowmin)
			{
				next=i;
				nowmin=mingo[i-1];
//cout <<  now << i << "nowmin:" << nowmin << endl;
			}
		}

		now = next;
		search[next-1]=true;
	}

	printf("%.10Lf\n",mingo[n+2-1]);
//	cout << mingo[n+2-1] << endl;
}
